odxtools/standardlengthtype.py

```diff
@@ -18,24 +20,47 @@ class StandardLengthType(DiagCodedType):
     def dct_type(self) -> DctType:
         return "STANDARD-LENGTH-TYPE"
 
-    def convert_internal_to_bytes(self, internal_value, encode_state: EncodeState,
+    def __post_init__(self):
+        if self.bit_mask is not None:
+            maskable_types = (DataType.A_UINT32, DataType.A_INT32, DataType.A_BYTEFIELD)
+            odxassert(
+                self.base_data_type in maskable_types,
+                'Can not apply a bit_mask on a value of type {self.base_data_type}',
+            )
+
+    def __apply_mask(self, internal_value: AtomicOdxType) -> AtomicOdxType:
+        if self.bit_mask is None:
+            return internal_value
+        if self.is_condensed_raw is True:
+            raise NotImplementedError("Serialization of condensed bit mask is not supported")
+        if isinstance(internal_value, int):
+            return internal_value & self.bit_mask
+        if isinstance(internal_value, bytes):
+            int_value: int = int.from_bytes(internal_value, 'big')
```

Comment:
the explicit type specification is not necessary here: mypy deduces the type of `int_value` correctly, and by convention, `.from_*()` constructors return an object of the type it was called for. That said, if you prefer to leave it in, that's perfectly fine with me...

Reply:
vscode was acting a bit weird, started behaving correctly when I added it

```suggestion
            int_value = int.from_bytes(internal_value, 'big')
```